qt/widgets/common/src/ImageInfoWidget.cpp

```diff
@@ -0,0 +1,79 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidQtWidgets/Common/ImageInfoWidget.h"
+#include "MantidAPI/Workspace_fwd.h"
+#include "MantidQtWidgets/Common/ImageInfoModelMD.h"
+#include "MantidQtWidgets/Common/ImageInfoModelMatrixWS.h"
+
+#include <QAbstractScrollArea>
+#include <QHeaderView>
+#include <QTableWidget>
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+/**
+ * Constructor
+ */
+ImageInfoWidget::ImageInfoWidget(const Mantid::API::Workspace_sptr &workspace,
+                                 QWidget *parent)
+    : QTableWidget(2, 0, parent) {
+  createImageInfoModel(workspace);
+  horizontalHeader()->hide();
+  verticalHeader()->hide();
+
+  int height = 2;
+  for (int i = 0; i < rowCount(); i++)
+    height += rowHeight(i);
+
+  setFixedHeight(height);
```

Comment:
So I did this because there was a lot of white space around the table but I've changed it so rather than it being a fixed height it adjusts it's size based on it contents and it works much better.